internal/service/alertconfiguration/data_source_alert_configuration.go

```diff
@@ -438,16 +439,14 @@ func convertNotificationToCtyValues(notification *admin.AlertsNotificationRootFo
 		values["sms_enabled"] = cty.BoolVal(*notification.SmsEnabled)
 	}
 
-	if len(notification.Roles) > 0 {
-		roles := make([]cty.Value, 0)
-
-		for _, r := range notification.Roles {
+	if roles := notification.GetRoles(); len(roles) > 0 {
+		roleList := make([]cty.Value, 0)
```

Comment:
can this be pre-allocated?
```suggestion
		roleList := make([]cty.Value, 0, len(roles))
```

Reply:
in this PR i'm trying to change as few things as possible, you're right there are many loops than can be improved, but I prefer to do that in next PRs

Reply:
SGTM